### PR TITLE
Added cmake and CI support

### PR DESCRIPTION
**Pull request by winterheart**
Added CMake and CI (based on Github Actions) support

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,5 +1,8 @@
 ## QUASI88改 / QUASI88kai
 ---
+
+![](https://github.com/rzumer/quasi88/workflows/build/badge.svg)
```
Comment (rzumer) [inline README.md]: This should be a relative URL so forks can also display their build status. If I remember correctly it should work by setting the URL to `workflows/build/badge.svg`.

File: cmake/FindReadline.cmake [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+	VERSION_VAR
+	READLINE_VERSION
+	)
+
+if ( READLINE_FOUND )
+	set( READLINE_LIBRARIES "${READLINE_LIBRARY}" "${NCURSES_LIBRARY}" )
+	set( READLINE_INCLUDE_DIRS "${READLINE_INCLUDE_DIR}" )
+endif ()
+
+mark_as_advanced( READLINE_ROOT_DIR READLINE_INCLUDE_DIR READLINE_LIBRARY NCURSES_LIBRARY )
```
Comment (rzumer) [inline cmake/FindReadline.cmake]: Please add a trailing newline for consistency.

File: README.cmake.md [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
++-----------------+-----------------------------------+-----|
+| ENABLE_GTK2     | Enable GTK2 backend               | OFF |
+| ENABLE_SDL      | Enable SDL backend                | OFF |
+| ENABLE_X11      | Enable X.Org backend              |  ON |
+| ENABLE_JOYSTICK | Enable joystick support for X.Org (available options: NO, SDL, LINUX_USB, BSD_USB) | NO |
+| ENABLE_SOUND    | Enable sound support              |  ON |
+| ENABLE_FMGEN    | Enable FM sound generator         |  ON |
+| ENABLE_DOUBLE   | Enable double screen              |  ON |
+| ENABLE_UTF8     | Enable UTF-8                      |  ON |
+| ENABLE_PC8801_KEYBOARD_BUG | Enable emulation of keyboard bug | OFF |
```
Comment (rzumer) [inline README.cmake.md]: Please pad other options in the table so that they are aligned with this one.

File: CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,372 @@
+cmake_minimum_required(VERSION 3.10)
+
+project(quasi88 VERSION 0.6.8)
+list(APPEND CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/cmake)
+
+# Reasonable default
+set(CMAKE_C_STANDARD 99)
+set(CMAKE_C_STANDARD_REQUIRED TRUE)
+
+option(ENABLE_GTK2 "Enable building for GTK2 (currently broken)" OFF)
```
Comment (rzumer) [inline CMakeLists.txt]: Is only the CMake build broken for GTK2, or is it broken in general?

If it is broken, why is it enabled in CI?

**Comment by winterheart** [comment]
@rzumer done.

File: README.md (hunk 1 above)
Comment (winterheart) [reply README.md]: It's only works on full URL. I can't figure out how make it work with relative URL.

Comment (rzumer) [reply README.md]: I figured it out and pushed a fix directly.

**Comment by rzumer** [comment]
Thanks!

**Pull request merged**